core/ir/aarch64/codec.txt

```diff
@@ -294,1336 +294,1347 @@ x----------xxxxx----------------  wx16       # W/X register (or WZR/XZR)
+0x101110xx1xxxxx100011xxxxxxxxxx  n   61         cmeq            dq0 : dq5 dq16 bhsd_sz
+0x001110xx100000100110xxxxxxxxxx  n   61         cmeq            dq0 : dq5 bhsd_sz
+0101111011100000100110xxxxxxxxxx  n   61         cmeq             d0 : d5
+01111110111xxxxx100011xxxxxxxxxx  n   61         cmeq             d0 : d5 d16
+0x001110xx1xxxxx001111xxxxxxxxxx  n   62         cmge            dq0 : dq5 dq16 bhsd_sz
+0x101110xx100000100010xxxxxxxxxx  n   62         cmge            dq0 : dq5 bhsd_sz
+0111111011100000100010xxxxxxxxxx  n   62         cmge             d0 : d5
+01011110111xxxxx001111xxxxxxxxxx  n   62         cmge             d0 : d5 d16
+0x001110xx1xxxxx001101xxxxxxxxxx  n   63         cmgt            dq0 : dq5 dq16 bhsd_sz
+0x001110xx100000100010xxxxxxxxxx  n   63         cmgt            dq0 : dq5 bhsd_sz
+0101111011100000100010xxxxxxxxxx  n   63         cmgt             d0 : d5
+01011110111xxxxx001101xxxxxxxxxx  n   63         cmgt             d0 : d5 d16
+0x101110xx1xxxxx001101xxxxxxxxxx  n   64         cmhi            dq0 : dq5 dq16 bhsd_sz
+01111110111xxxxx001101xxxxxxxxxx  n   64         cmhi             d0 : d5 d16
+0x101110xx1xxxxx001111xxxxxxxxxx  n   65         cmhs            dq0 : dq5 dq16 bhsd_sz
+01111110111xxxxx001111xxxxxxxxxx  n   65         cmhs             d0 : d5 d16
+0111111011100000100110xxxxxxxxxx  n   66         cmle             d0 : d5
+0x101110xx100000100110xxxxxxxxxx  n   66         cmle            dq0 : dq5 bhsd_sz
+0x001110xx100000101010xxxxxxxxxx  n   67         cmlt            dq0 : dq5 bhsd_sz
+0101111011100000101010xxxxxxxxxx  n   67         cmlt             d0 : d5
+0x001110xx1xxxxx100011xxxxxxxxxx  n   68        cmtst            dq0 : dq5 dq16 bhsd_sz
+01011110111xxxxx100011xxxxxxxxxx  n   68        cmtst             d0 : d5 d16
+0x00111000100000010110xxxxxxxxxx  n   69          cnt            dq0 : dq5
+00011010110xxxxx010000xxxxxxxxxx  n   70       crc32b             w0 : w5 w16
+00011010110xxxxx010100xxxxxxxxxx  n   71      crc32cb             w0 : w5 w16
+00011010110xxxxx010101xxxxxxxxxx  n   72      crc32ch             w0 : w5 w16
+00011010110xxxxx010110xxxxxxxxxx  n   73      crc32cw             w0 : w5 w16
+10011010110xxxxx010111xxxxxxxxxx  n   74      crc32cx             w0 : w5 x16
+00011010110xxxxx010001xxxxxxxxxx  n   75       crc32h             w0 : w5 w16
+00011010110xxxxx010010xxxxxxxxxx  n   76       crc32w             w0 : w5 w16
+10011010110xxxxx010011xxxxxxxxxx  n   77       crc32x             w0 : w5 x16
+x0011010100xxxxxxxxx00xxxxxxxxxx  r   78         csel            wx0 : wx5 wx16 cond
+x0011010100xxxxxxxxx01xxxxxxxxxx  r   79        csinc            wx0 : wx5 wx16 cond
+x1011010100xxxxxxxxx00xxxxxxxxxx  r   80        csinv            wx0 : wx5 wx16 cond
+x1011010100xxxxxxxxx01xxxxxxxxxx  r   81        csneg            wx0 : wx5 wx16 cond
+110101010000100001111110010xxxxx  n   569     dc_cisw                : x0
```

Comment:
OK it wasn't easy but I found what really changed: these were inserted.  So this was possible w/o changing to xml as #4393 talked about?  Not even a change to the python script was needed?

Reply:
> So this was possible w/o changing to xml as #4393 talked about? Not even a change to the python script was needed?

You mean why this PR doesn't make any python script changes to ensure ordering? The ordering related stuff was already handled by 2c4f1e5a30a98ba3061e78d358ed3b60377f8ce6, I think. IIUC this PR needs to only add new entries in the codec, which is what it does.

Reply:
#4393 was filed about scrapping this codec input format and switching to xml input in order to provide enough flexibility for splitting SYS into sub-opcodes.  But here it seems we're able to do that split in this codec format.  Was it a prior augmentation that gave us that capability?

Reply:
> So this was possible w/o changing to xml as #4393 talked about?

Yes, the comments in #4393 were optimistic about using XML for v8.0 and the comment about SYS/DC/IC was more of an example of what could be done rather than what we would do in the current codec. The plan is to use XML for Neoverse V1 and N2.

> Not even a change to the python script was needed?

No change to the script, but `codec.txt` change has to be run through `codecsort.py` which re-orders the new instructions and in so doing can re-align fields which results in space changes. Not ideal but I use the `Hide whitespace` option in the `Diff view` confing on GitHub for `codec.txt`.

> The ordering related stuff was already handled by 2c4f1e5, I think. IIUC this PR needs to only add new entries in the codec, which is what it does.

Yes.

As an aside, we need to review the approach proposed by #4393 and decide how to make the most of automating future instruction support. That is a discussion for #4393 we should have in January 2022, or offline as it is a siginificant decision with relatively long-term consequences.

Reply:
Ack.